packages/promise-kit/src/promiseKit.js

```diff
@@ -49,19 +50,20 @@ export function makePromiseKit() {
   // Node.js adds the `domain` property which is not a standard
   // property on Promise. Because we do not know it to be ocap-safe,
   // we remove it.
-  if (p.domain) {
+  if ('domain' in p) {
```

Comment:
Please double-check with @erights. Getting this wrong has security implications. `p.domain`, `p.domain !== undefined`, and `'domain' in p` are not exactly equivalent. I suspect that this is the best of breed since it is strictest, not even allowing a falsy property if it’s present. I’m not sure how `in` interacts with getters and setters, but maybe it would interfere with a taming?

Reply:
`in` doesn't invoke getters, only checks for property presence anywhere on the prototype chain, so the check should strictly be true more often than before.

Reply:
> in doesn't invoke getters, only checks for property presence anywhere on the prototype chain, so the check should strictly be true more often than before.

Yes. This is better. In fact though, this entire bit of defensiveness adds no protection against malice and little protection against accident. We need to get node to stop adding `domain`. Until we do we cannot be secure on node.